### PR TITLE
NEX-44: Add working webform example

### PR DESCRIPTION
**Pull request by vermario**
This branch adds a simple working webform example in backend and frontend.

We decided (for now at least) to implement a static form, in the sense that the "hardcoded" fields in the frontend need to match the fields in the webform in the backend. Creating a dynamic system of forms that automatically match what's coming from the user-defined webforms in Drupal is out of scope for this template.

On the backend:
* Added webform and webform_rest
* added permission for anonymous user to post to webform

On the frontend:
* added an API route for the form to post to, which in turn will post to drupal
* added a form component with fields matching the fields defined in the default contact webform on the drupal side.
* Added translations for the user interface of the form
* Added hook_react_form to be able to reset the form and better handle it. I have not added validation since the html5 validation handles this basic case. I have added a thank you message with a button to submit the form again.

### Testing instructions

1. run the megacommand to setup the project from scratch https://github.com/wunderio/next4drupal-project#%EF%B8%8F-quick-one-command-setup
2.  visit frontpage, submit form, see thank you message, click button to submit again, see form again
3.  log into backend `lando drush uli`
4.  visit backend at: https://next4drupal-project.lndo.site/en/admin/structure/webform/manage/contact/results/submissions  and you should find your submission 

### Screenshots
Form on frontpage:
image

Confirmation message + button to reset form:
image

Submission data in backend:
image

### CODE REVIEW

File: next/.eslintrc.js [hunk 1]
```diff
@@ -22,6 +22,14 @@ module.exports = {
   ],
   rules: {
     "prettier/prettier": "error",
+    "@typescript-eslint/no-misused-promises": [
```
Comment (vermario) [inline next/.eslintrc.js:25]: @joshua-scott  I found this discussion on hook_react_form's repository: https://github.com/react-hook-form/react-hook-form/discussions/8622 

It seems that it's accepted there that this is the recommended way to fix the issue of "Promise-returning function provided to attribute where a void return was expected."  so I went with this.

**Comment by hkirsman** [comment]
Wha'ts the megacommand? This https://github.com/wunderio/next4drupal-project#%EF%B8%8F-quick-one-command-setup ?

**Comment by vermario** [comment]
> Wha'ts the megacommand? This https://github.com/wunderio/next4drupal-project#%EF%B8%8F-quick-one-command-setup ?

Yes! :) We have been calling it that

**Comment by hkirsman** [comment]
> > Wha'ts the megacommand? This https://github.com/wunderio/next4drupal-project#%EF%B8%8F-quick-one-command-setup ?
> 
> Yes! :) We have been calling it that

What about `lando reset` like in tcom? :)

**Comment by hkirsman** [comment]
That's a bit odd, that it's building production with the megacommand:

**Comment by vermario** [comment]
> What about `lando reset` like in tcom? :)

that could work! sometimes we also want to do lando rebuild (for example when adding services etc) but that's rare... I will add it!

**Comment by vermario** [comment]
> That's a bit odd, that it's building production with the megacommand. Taking some time..
The reason for this is that we don't have a hosted site anywhere for now, and there are some errors (typescript) that only happen when you do `npm build` so they could go unnoticed if we only did npm run dev in that command.

**Comment by hkirsman** [comment]
> > That's a bit odd, that it's building production with the megacommand. Taking some time..
> 
> The reason for this is that we don't have a hosted site anywhere for now, and there are some errors (typescript) that only happen when you do `npm build` so they could go unnoticed if we only did npm run dev in that command.

I think there was also something off as it took more than an hour. Trying again.

**Review by joshua-scott — APPROVED** [review]
This is working very nicely and is a good example for this template. I pushed one small commit to pass those "required" options to react-hook-form (df6e4f93dd473ea7a331e9a27f037c8030a72fb8 - doesn't make a difference yet but can if more validation is added later.)

**Pull request merged**